headers/types/common/common.h

```diff
@@ -1230,4 +1231,102 @@ struct monster_sprite_data_entry {
 };
 ASSERT_SIZE(struct monster_sprite_data_entry, 2);
 
+// Holds basic information about a monster.
+struct monster_data_table_entry {
+    uint16_t entity_id; // 0x0: A entity-unique ID to identify the entity in both of its 2 entries.
+                        // (Independent from file index)
+    uint16_t
+        unk_0x2; // 0x2: Unknown. Pokemon with the same category string have the same value in here.
+    uint16_t pokedex_number; // 0x4: The national Pokedex number, as displayed in Chimecho Assembly.
+    uint16_t base_movement_speed; // 0x6: The base movement speed in dungeons.
+    uint16_t pre_evolution_idx; // 0x8: The INDEX of the pre-evolution inside the "monster.md" file.
+                                // Not the Pokedex number, or the Entity ID! Is 0 if none.
+    uint16_t evolution_method;  // 0xA: The evolution method required to evolve to this Pokemon from
+                                // the pre-evo specified in PreEvoIndex. Null if unused.
+    uint16_t
+        evolution_param_1; // 0xC: The first parameter for the evolution method. Null if unused.
+    uint16_t
+        evolution_param_2; // 0xE: The second parameter for the evolution method. Null if unused.
+    uint16_t sprite_index; // 0x10: The index of the entity's sprite. It's the index inside the
+                           // three Pokemon sprite files inside the "/MONSTER/" directory!
+    struct monster_gender_8 gender; // 0x12: Gender of this particular entity entry.
+    uint8_t body_size; // 0x13: The body size of the Pokemon. Used when determining how many Pokemon
+                       // fits in the party.
+    struct type_id_8 primary_type;        // 0x14: The Pokemon's primary type.
+    struct type_id_8 secondary_type;      // 0x15: The Pokemon's secondary type.
+    struct mobility_type_8 movement_type; // 0x16: This decides what terrains the entity can move
+                                          // over, and whether its hovering or not.
+    struct iq_group_id_8 iq_group;        // 0x17: The IQ group the Pokemon belongs to.
+    uint8_t primary_ability;              // 0x18: The Pokemon's primary ability's ID.
+    uint8_t secondary_ability;            // 0x19: The Pokemon's secondary ability's ID.
+    // 0x1A: A bitfield. Most likely used to store bitflags.
```

Comment:
- Combine consecutive unknown fields into one.
- We can combine the second byte into the first one in a single `uint16_t`
- Prefix known bool fields with `f_`.

Something like:

```c
uint16_t unk_flags_0 : 3
bool f_can_move : 1
...
uint16_t unk_flags_8 : 8
```